trin-core/src/types/bus.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::anyhow;
+use tokio::sync::mpsc;
+
+use crate::jsonrpc::types::{HistoryJsonRpcRequest, StateJsonRpcRequest};
+
+/// Datatype to house all jsonrpc tx channels for available subnetworks
+#[derive(Clone, Debug, Default)]
+pub struct NetworkBus {
+    pub history_jsonrpc_tx: Option<mpsc::UnboundedSender<HistoryJsonRpcRequest>>,
+    pub state_jsonrpc_tx: Option<mpsc::UnboundedSender<StateJsonRpcRequest>>,
+}
+
+impl NetworkBus {
+    pub fn get_history_tx(&self) -> anyhow::Result<mpsc::UnboundedSender<HistoryJsonRpcRequest>> {
+        match self.history_jsonrpc_tx.clone() {
+            Some(val) => Ok(val),
+            None => Err(anyhow!("History subnetwork is not available")),
+        }
+    }
```

Comment:
The reason I went with `Result` is that it helps clean up the code that gets the tx. If we return a result, we can just unwrap it using the `?` operator, and let anyhow bubble the error up. If we return an `Option`, then it requires a slightly more complex unwrapping / error handling. Also, since the attribute itself is an `Option`, then we wouldn't really need these getter functions, since we could just access it directly. But the whole point of this struct is to help clean up the code in other parts of the repo where we get / set these channels.

Reply:
I see your point, I think prefixing the function name with `try_` is reasonable and this indicates that the return type is a `Result`.

Reply:
IMO we should not return `Result` here to make it more convenient for the caller. The caller should be responsible for converting the value returned by the function, and that should be straightforward in this instance. For example:

```rust
let tx = bus.get_history_tx().ok_or(Error::Foo)?;
```